application/dash_application/utility/plot_builder.py

```diff
@@ -309,11 +389,72 @@ def generate(title_text, sorted_data, x_fn, y_fn, axis_text, colourby, shapeby,
                 'title': {
                     'text': axis_text
                 }
+            },
+            legend={
+                'tracegroupgap': 0
             }
         )
     )
 
 
+def generate_subplot(subplots: List[Subplot]):
+    """
+    Generates a subplot using functions that take a DataFrame and graph paramaters,
+    returning a list of traces.
+
+    """
+    fig = make_subplots(
+        rows=len(subplots),
+        cols=1,
+        vertical_spacing=0.02,
+        shared_xaxes=True,
+        subplot_titles=[subplot.title for subplot in subplots]
```

Comment:
Ah, I see now why the titles couldn't have gone in the generate_*_subplot functions. What a strange decision on plotly's part